src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2018,6 +2018,62 @@ public async Task<IOperationResult> RevertRevisions(DateTime before, TimeSpan wi
             return await RevertRevisions(before, window, onProgress, collections: null, token);
         }
 
+        public async Task<IOperationResult> RevertRevisions(Action<IOperationProgress> onProgress, string id, string cv,
+            OperationCancelToken token)
+        {
+            var result = new RevertResult();
+
+            if (id == null)
+                throw new ArgumentException("Document id is null");
+
+            if (id == string.Empty)
+                throw new ArgumentException("Document id is an empty string");
+
+            if (cv == null)
+                throw new ArgumentException("Change Vector is null");
+
+            if (id == string.Empty)
+                throw new ArgumentException("Change Vector is an empty string");
+
+
+            using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+                using (Slice.From(context.Allocator, cv, out var cvSlice))
+                {
+                    if (table.ReadByKey(cvSlice, out TableValueReader tvr) == false)
+                    {
+                        var msg = $"There is no revision with the cv {cv}";
+                        if (_logger.IsInfoEnabled)
+                            _logger.Info(msg);
+                        result.WarnAboutFailedDoc(msg);
```

Comment:
Why are you reporting this rather than throwing?

Reply:
Done (now we are throwing)